### DISCUSSION TITLE
Somehow I broke the main page's CSS, but only for safari and `public/index.html`

### DISCUSSION THREAD

**Discussion by MattDodsonEnglish** [thuliteio/doks#258] [Q&A]
First of all, let me say thanks. It's been a joy to learn Hugo with your template. We've shown a few users our new docs site, and the first thing everyone has said is "Wow, it looks amazing." I think this theme will take off.

On a safari browser, someone noted that all styling was broken for the blog and homepage on our netlify deployment. It worked fine on all other browsers. I've heard that safari is stricter than other browsers, so maybe it related to that.

So I ran `hugo` and I found that sure enough, CSS was broken for `public/index.html` too. I have no idea how I broke the CSS but, I guess its related the files in `resources/_gen/`.  Unfortunately I don't really understand what's happening though.

Does anyone have any idea how I can troubleshoot or fix this? Apparently the netlify page is working on Safari now, but the pages in `public` still are broken.

Thanks again.

For reference:
-our deployment is here: https://trueblocks.netlify.app/docs/prologue/introduction/
-our repo is here: https://github.com/TrueBlocks/trueblocks-docs

**Comment by MattDodsonEnglish** [comment]
Hey @h-enk , I just reproduced this on the actual getdpks.org site too. Docs and Tutorial have no styling. The others look as expected.

This screenshot is from 5 minutes ago, on an ipad with safari.

![AAFC27F5-318F-4663-BC39-484EB4135E1A](https://user-images.githubusercontent.com/47385188/116791330-de351c80-aa8f-11eb-9961-72ae8dd557e2.jpeg)

**Reply by h-enk** [reply]
Hmm, strange... Unfortunately, I cannot verify this myself — I'm not an Apple user.

Would you mind checking if any Content Security Policy errors are triggered /or any other console errors?

**Reply by h-enk** [reply]
I got hold on a recent iPad Pro w/ latest Safari version. Cannot confirm what you experienced /getdoks.org (incl. tutorial + docs trees) shows as it should to me.

Anyone else experiencing this behavior?

P.S. I assume you've cleared Safari's cache?

**Reply by MattDodsonEnglish** [reply]
Unfortunately I only have occasional access to an ipad also. I didn't clear the cache, but when I switched off "Content Blockers," your site was working. I think you need to plug an Ipad into a mac computer to use its console, so I couldn't do that.

However, my colleague, who does use safari, did notice that styles were lost on our netlify deployment on his mac safari browser. But only for a moment. You can see his comment in the issue I made on our repo: TrueBlocks/trueblocks-docs#14 (comment)

**Comment by MattDodsonEnglish** [comment]
I am still struggling with getting the site to work from `public` on my local machine. I think it's related to my the way I've configured `baseURL` in config. But I've tried a few different combinations, and I'm still getting the same three errors.

In firefox's `about:config`, I  disabled `privacy.file_unique_origin` which should allow local file loading. 

These are the errors with `baseURL = ''`. By the way, it works fine running `hugo server` , only public is causing a problem. 

Do you have any ideas?

```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at file:///index.min.f88f1d16b81b7b7724d13a41b67b2e42773912feaacde55cb4237178390eefdd23b9dae0e375ade709e0514b0c973124ed5f4752c0f8b1488518dea7f4e385bb.js. (Reason: CORS request not http).

<script> source URI is not allowed in this document: “file:///index.min.f88f1d16b81b7b7724d13a41b67b2e42773912feaacde55cb4237178390eefdd23b9dae0e375ade709e0514b0c973124ed5f4752c0f8b1488518dea7f4e385bb.js”. index.html:187:1

The resource at “file:///fonts/vendor/jost/jost-v4-latin-regular.woff2” preloaded with link preload was not used within a few seconds. Make sure all attributes of the preload tag are set correctly.
```

**Reply by h-enk** [reply]
@MattDodsonEnglish, still having issues?

I ran a local development + production build of https://github.com/TrueBlocks/trueblocks-docs on my Windows machine w/ no issues.

**Reply by MattDodsonEnglish** [reply]
Hey @h-enk 

Thanks so much for testing this! I think we can close this discussion. I think my problem is some weird edge case, which it's my responsibility to solve.
 
I can confirm that running hugo server works fine on linux + firefox/chrome, and on my colleague's Mac + safari. Unfortunately, neither of us can make the build in the  `public` directory work.  But we're just going to move on from this; if we ever do serve the site from our own server, we'll figure out what we need to do.

Thanks so much for your hard work Henk! I recently shared the Doks theme on the "write the docs" technical writers Slack channel, and it got a lot of positive attention.

**Reply by h-enk** [reply]
> I recently shared the Doks theme on the "write the docs" technical writers Slack channel, and it got a lot of positive attention.

Thanks for that!
